### PR TITLE
Deduplicate test-result attestations by test name

### PR DESCRIPTION
**Pull request by joejstuart**
When a test task is re-run, each execution produces a separate test-result attestation. Previously all attestations were evaluated, so a stale FAILED result would block even after a successful re-run.

Group attestations by test name (predicate.configuration[0].name) and keep only the one with the latest predicate.timestamp. Attestations without a timestamp are excluded from evaluation entirely.

[EC-2082](https://redhat.atlassian.net/browse/EC-2082)

### CODE REVIEW

**Review by st3penta — APPROVED** [review]
LGTM.
The only thing worth fixing is maybe the fail-open finding by fullsend, unless we are sure that all the konflux tests generate the test attestations in the same format (in particular the `name` and `timestamp` fields). If that's the case, it should be impossible to bypass the filter, and it's good as is.

**Comment by joejstuart** [comment]
> LGTM. The only thing worth fixing is maybe the fail-open finding by fullsend, unless we are sure that all the konflux tests generate the test attestations in the same format (in particular the `name` and `timestamp` fields). If that's the case, it should be impossible to bypass the filter, and it's good as is.

All test attestations should produce those fields. Also, we have a story to require certain test attestations so that would throw a violation.

File: policy/release/test_attestation/test_attestation.rego [hunk 1]
```diff
@@ -34,7 +34,34 @@ import data.lib.json as j
...
+		}
+	}
+
+	# For each group, find max timestamp once
+	some name, statements in grouped
+	max_ts := max({_attestation_timestamp(s) | some s in statements})
+
+	# Filter to latest
+	some statement in statements
+	_attestation_timestamp(statement) == max_ts
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply policy/release/test_attestation/test_attestation.rego:61]: This edge case (multiple attestations with identical timestamps) is extremely unlikely in practice since Konflux test attestations include nanosecond-precision RFC3339 timestamps from different pipeline runs. The current behavior (keeping all attestations with the max timestamp) is actually fail-closed: if any of those equal-timestamp attestations has a failed result, the release will still be blocked by the  rule.

While deterministic tie-breaking would be more precise, the added complexity isn't warranted given:
1. All Konflux test attestations produce valid timestamps
2. There's a separate story to require certain test attestations (which would catch missing/invalid ones)
3. The current behavior is already fail-closed (multiple attestations with max timestamp = all evaluated)

Resolving as won't fix.
